src/components/Player.js

```diff
@@ -24,38 +20,31 @@ const Player = ({
     animationPercentage: 0,
   });
   const [songVolume, setSongVolume] = useState(50);
-
-  const activeLibraryHandler = (nextPrev) => {
-    const newSongs = songs.map((song) => {
-      if (song.id === nextPrev.id) {
-        return {
-          ...song,
-          active: true,
-        };
-      } else {
-        return {
-          ...song,
-          active: false,
-        };
-      }
-    });
-    setSongs(newSongs);
-  };
+  const [isPlaying, setIsPlaying] = useState(false);
+  const audioRef = useRef(null);
 
   // Event Handlers
   const playSongHandler = () => {
-    if (isPlaying === false) {
-      audioRef.current.play();
+    if (!isPlaying) {
       setIsPlaying(!isPlaying);
 
       //control volume
       audioRef.current.volume = 0.1;
     } else {
-      audioRef.current.pause();
       setIsPlaying(!isPlaying);
     }
   };
 
+  /* Tutaj bardzo upraszający wszystko kod. We wszystkich miejscach w aplikacji wystarczy 
+     ze będziesz operował isPlaying, a ten useEffect będzie odpalał/zatrzymywał player zależnie od jego wartości. */
+  useEffect(()=> {
+    if(isPlaying) {
+      audioRef.current.play();
+    }else {
+      audioRef.current.pause();
+    }
+  },[currentSong, isPlaying]);
+
```

Comment:
Tak, tutaj jeśli jeszcze się nie spotkałeś z `useEffect`, to polecam np. na oficjalnej stronie sobie obczaić o co chodzi. W skrócie ten hook daje możliwość zareagowania na 1. render + każdą zmianę wartości z tabeli zależności (2. argument hooka).
W tym przypadku jeden fragment kodu obsługuje:
- Kliknięcie na przycisk play/pause
- Skipnięcie piosenki w Playerze
- Wybranie innej piosenki z Library

Dzięki temu audioRef jest tylko tam, gdzie rzeczywiście powinien być używany, co inne komponenty obchodzi czy player jest włączony czy nie, one odpowiadają za co innego.